SearchCancellationIT#testCancelFailedSearchWhenPartialResultDisallowed fails

**Build scan**: https://gradle-enterprise.elastic.co/s/khmvuaugydbz2/console-log?task=:server:internalClusterTest

**Repro line**: gradlew ':server:internalClusterTest' --tests "org.elasticsearch.search.SearchCancellationIT.testCancelFailedSearchWhenPartialResultDisallowed" -Dtests.seed=B02E83C609386A57 -Dtests.security.manager=true -Dtests.locale=en-SG -Dtests.timezone=PRT -Druntime.java=11

**Reproduces locally?**: No

**Applicable branches**: master

**Failure history**: newly introduced test, see #63520

**Failure excerpt**:

```
org.elasticsearch.search.SearchCancellationIT > testCancelFailedSearchWhenPartialResultDisallowed FAILED |  
-- | --
  | java.lang.AssertionError: |  
  | Expected: a collection with size <1> |  
  | but: collection size was <0> |  
  | at __randomizedtesting.SeedInfo.seed([B02E83C609386A57:FA326EC8B97B8D6F]:0) |  
  | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  | at org.junit.Assert.assertThat(Assert.java:956) |  
  | at org.junit.Assert.assertThat(Assert.java:923) |  
  | at org.elasticsearch.search.SearchCancellationIT.lambda$testCancelFailedSearchWhenPartialResultDisallowed$5(SearchCancellationIT.java:327) |  
  | at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:939) |  
  | at org.elasticsearch.search.SearchCancellationIT.testCancelFailedSearchWhenPartialResultDisallowed(SearchCancellationIT.java:325)

```

--- a/server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java
@@ -281,7 +281,6 @@ public class SearchCancellationIT extends ESIntegTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/63976")
     public void testCancelFailedSearchWhenPartialResultDisallowed() throws Exception {
         final List<ScriptedBlockPlugin> plugins = initBlockFactory();
         int numberOfShards = between(2, 5);
@@ -325,8 +324,12 @@ public class SearchCancellationIT extends ESIntegTestCase {
             queryLatch.countDown();
             assertBusy(() -> {
                 final List<SearchTask> searchTasks = getSearchTasks();
-                assertThat(searchTasks, hasSize(1));
-                assertTrue(searchTasks.get(0).isCancelled());
+                // The search request can complete before the "cancelledLatch" is latched if the second shard request is sent
+                // after the request was cancelled (i.e., the child task is not allowed to start after the parent was cancelled).
+                if (searchTasks.isEmpty() == false) {
+                    assertThat(searchTasks, hasSize(1));
+                    assertTrue(searchTasks.get(0).isCancelled());
+                }
             }, 30, TimeUnit.SECONDS);
         } finally {
             for (ScriptedBlockPlugin plugin : plugins) {

